core/src/main/java/org/infinispan/interceptors/locking/OptimisticLockingInterceptor.java

```diff
@@ -60,6 +60,11 @@ public Object visitPrepareCommand(TxInvocationContext ctx, PrepareCommand comman
       final Collection<Object> keysToLock = command.getKeysToLock();
       ((TxInvocationContext<?>) ctx).addAllAffectedKeys(command.getAffectedKeys());
       if (!keysToLock.isEmpty()) {
+         if (command.isRetriedCommand() && ctx.isOriginLocal()) {
```

Comment:
Shouldn't we exclude `onePhaseCommit` prepares as well? At least with pessimistic locking, the prepare command isn't going to acquire any locks.